pkg/provisioner/submitChallenge.go

```diff
@@ -0,0 +1,212 @@
+/*
+ *
+ *  MIT License
+ *
+ *  (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a
+ *  copy of this software and associated documentation files (the "Software"),
+ *  to deal in the Software without restriction, including without limitation
+ *  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ *  and/or sell copies of the Software, and to permit persons to whom the
+ *  Software is furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included
+ *  in all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ *  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ *  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ *  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *  OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+package provisioner
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"log"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/cray-hpe/tpm-provisioner/third_party/devid-provisioning-tool/pkg/common"
+	"github.com/cray-hpe/tpm-provisioner/third_party/devid-provisioning-tool/pkg/devid"
+	"github.com/cray-hpe/tpm-provisioner/third_party/devid-provisioning-tool/pkg/x509tcg"
+)
+
+// SubmitResponse contains the response to the submit challenge api request.
+type SubmitResponse struct {
+	Success          bool   `json:"success"`
+	Reason           string `json:"reason,omitempty"`
+	DevIDCertificate string `json:"devIdCertificate"`
+}
+
+// SubmitRequest contains the request for the submit challenge api.
+type SubmitRequest struct {
+	Data string `json:"data"`
+}
+
+// SubmitChallenge handles the challenge/submit api request.
```

Comment:
Nit, another place that could probably use a bit of documentation re: application (really methods in this file in total)